src/sage/sets/recursively_enumerated_set.pyx

```diff
@@ -9,25 +9,23 @@ sets that are described by some ``seeds`` and a successor function
 graded, forest) or not. The elements of a set having a symmetric, graded or
 forest structure can be enumerated uniquely without keeping all of them in
 memory. Many kinds of iterators are provided in this module: depth first
-search, breadth first search or elements of given depth.
+search, breadth first search and elements of given depth.
 
 See :wikipedia:`Recursively_enumerable_set`.
 
-See documentation of :func:`RecursivelyEnumeratedSet` below for the
+See the documentation of :func:`RecursivelyEnumeratedSet` below for the
 description of the inputs.
 
 AUTHORS:
 
 - Sébastien Labbé, April 2014, at Sage Days 57, Cernay-la-ville
 
-EXAMPLES:
-
```

Comment:
This should not be removed. What follows are examples.

Reply:
If you look at the html, it looks terrible - and quite confusing.
<img width="960" alt="image" src="https://github.com/sagemath/sage/assets/8470093/1fe306ca-5889-41cd-bb84-311ff3d5e26e">

Reply:
Well, when I saw it I thought that for some reason (eg., bad markup) the examples were not typeset.

Would it be OK for you if we add instead `EXAMPLES::` right before the examples in each subsection?

I thought that the `EXAMPLES::` markup is only used in docstrings of classes, methods and functions, not at module level, and I could not find file where we have examples at module level interoduced with `EXAMPLES::`.

But in any case, I agree that the sizing of the headings is not good.  Do you have a best-practice example with a similar structure (i.e., with a guided tour at module level)?  I only know of `bijectionist.py`, `findstat.py`, `growth.py`, but they have all been written by myself.

Reply:
The whole list are the examples broken down into different sections. We do have other examples scattered around, but they are less frequent as the class-level doc is usually a better place for `?` access (e.g., `libgap_group.py` has some module-level doc examples, but likely not the best place). So I would prefer to keep it unchanged.

We don't really have a best practice, but perhaps `.. RUBRIC::`? H4 level might be fine too.

Reply:
Okay,
I have added `EXAMPLES::` just before each of the examples. Please kindly have a look at it and suggest any changes if required.

Reply:
@Omegaconstant Please revert this. As I said, each block is part of the examples, grouped accordingly.

Reply:
I disagree here. We have several sections, each with its own examples. Why are you against this?

Please take note that the original had me thinking that there would be a mistake.

Reply:
Because it is not describing the class/structure in each section. It is dividing the examples by section. By putting `EXAMPLES::` for each section, it no longer makes sense because there is nothing within each section but examples. In fact, it even has two `Forest Structure` examples (with the latter one having a redundant "Example:" included). The original version is just displayed larger than it should, which might not have been as such on previous versions too.

Reply:
Oops,
I just reverted. :sweat:
Apologies for the confusion. Please let me know what is to be done here.

Actually, in the original doc, the `EXAMPLES::` tag exists pretty much all the places just before the examples, except for the initial few paragraphs/ sections.

Reply:
OK, I don't like it at all, but I don't care enough.

For me, "EXAMPLES:" looks like a heading.  I would say that, ordinarily, a heading can only be followed by a sub-heading - however, "EXAMPLES:" is always just in capital letters, which would be the smallest possible heading.

Reply:
In my opinion,
The initial sections including `No hypothesis on the structure`, `Symmetric structure`, `Graded structure` and `Forest structure` -- each of these has a brief description regarding the structure followed by examples (just before which I added the `EXAMPLES::` tag). In the Forest structure section, there are three consecutive examples, I removed the large section heading of `Example: Forest structure` and `Example: Forest structure 2` and added a single tag `EXAMPLES::` before the start of the examples just after the forest structure.

The inclusion of the `EXAMPLES::` tag just before the example makes more sense than adding a single tag at the beginning of the entire document just after the name of the author and just before the `No hypothesis on the structure`.

I hope this is fine.

Reply:
> In my opinion, The initial sections including `No hypothesis on the structure`, `Symmetric structure`, `Graded structure` and `Forest structure` -- each of these has a brief description regarding the structure followed by examples (just before which I added the `EXAMPLES::` tag).

That's not generally true. I only see this for the "no hypothesis" section.

> In the Forest structure section, there are three consecutive examples, I removed the large section heading of `Example: Forest structure` and `Example: Forest structure 2` and added a single tag `EXAMPLES::` before the start of the examples just after the forest structure.
> 
> The inclusion of the `EXAMPLES::` tag just before the example makes more sense than adding a single tag at the beginning of the entire document just after the name of the author and just before the `No hypothesis on the structure`.

No, it doesn't. All of those that follow in the original `EXAMPLES:` are examples of the module that are explained in detail. They don't talk about the assumptions/structure needed to be satisfies (e.g., a formal definition) or have any generality.

As further evidence, your added `EXAMPLES::` is following after each block already described the example. This is clearly out of place (plus the English sentence syntax is wrong with the previous lines ending with the colon).

Either this needs to be reverted or the doc needs to be completely rewritten in a different format. We can address @mantepse's issue by replacing the header part with a smaller block; e.g,:
```diff
-Example: Forest structure
--------------------------
+.. RUBRIC:: Forest structure [Ex num]
```
That should make it look much better in the compiled doc.

Reply:
Cool.
I have understood your point and I have made the changes accordingly (as of now).
Actually, I did not look at the initial sections of the doc from this point of view (that -- the initial sections are the examples for some instances of the recursively enumerated set under some assumptions) and yes, they lack the proper formal definition. We should discuss about it.

Also, I am thankful to everyone for this discussion :-)